### PR TITLE
Set up continuous integration with github actions

### PR DESCRIPTION
**Pull request by leomartinez2020**
This PR implements continuous integration with github actions

### CODE REVIEW

File: .github/main.yml [hunk 1]
```diff
@@ -0,0 +1,22 @@
...
+      - name: Install nodejs
+        uses: actions/setup-node@v3
+        with:
+          node-version: '16'
+      - name: Install cypress
+        run: npm install cypress@latest
+      - name: Install linter
+        run: npm install eslint --save-dev
+      - name: Install linter cypress
+        run: npm install eslint-plugin-cypress --save-dev
```
Comment (holgiosalos) [inline .github/main.yml]: these installs are not needed, as the dependencies are already specified in package.json file, the only required step is to run the ci:
```
      - name: Install
        run: npm ci
```

File: .github/main.yml [hunk 2]
```diff
@@ -0,0 +1,22 @@
+name: GitHub Actions Demo
+on: [push]
```
Comment (holgiosalos) [inline .github/main.yml]: currently there are no jobs or checks executed for this PR, review if this `on` trigger needs to be changed by a:
`on: [pull_request]`. 

Also review that you have enabled the Require status check option in the branch protection rules:
image

**Comment by leomartinez2020** [comment]
@holgiosalos I added the necessary changes to `main.yml` and updated the branch rules as seen in the image:
cypress-ci

**Review by holgiosalos — COMMENTED** [review]
not sure, but, something is not well configured yet, since the checks tab, does not have active jobs running:
image

**Comment by leomartinez2020** [comment]
@holgiosalos <p><s>in that case I will have to create a new PR so that the check is triggered on pull_request as expected</s></p>

Update: I found the mistake: `main.yml` was not in the workflows folder. No need to create another PR

**Review by holgiosalos — APPROVED** [review]

**Pull request merged**